handlers/handler-loadbalance/src/main/java/org/apache/servicecomb/loadbalance/ServiceCombServer.java

```diff
@@ -79,29 +69,16 @@ public String getHost() {
     return endpoint.getEndpoint();
   }
 
-  public void clearContinuousFailure() {
-    continuousFailureCount.set(0);
-  }
-
-  public void incrementContinuousFailureCount() {
-    if (continuousFailureCount.get() < Integer.MAX_VALUE) {
-      continuousFailureCount.incrementAndGet();
-    }
-  }
-
-  public int getCountinuousFailureCount() {
-    return continuousFailureCount.get();
-  }
-
+  // take endpoints that belongs to same instance as same server
   public boolean equals(Object o) {
-    if (o instanceof CseServer) {
-      return this.getHost().equals(((CseServer) o).getHost());
+    if (o instanceof ServiceCombServer) {
```

Comment:
rest endpoint equals highway endpoint?
why?

Reply:
DiscoveryFilter sees MicroserviceInstance, not Endpoint. And the health check is also to MicroserviceInstance, not Endpoint.